### PR TITLE
Game over screen

### PR DESCRIPTION
**Pull request by justinvdk**
Adds end-game screen functionality. Also writes highscores into json file

### CODE REVIEW

**Comment by BCrlrd** [comment]
Please fix failing checks.

< review

**Comment by justinvdk** [comment]
My bad. didn't re-check all before merging my last changes onto my base branch

File: src/main/java/nl/tudelft/semgroup4/GameEndedState.java [hunk 1]
```diff
@@ -0,0 +1,232 @@
...
+
+                game.enterState(0);
+            }
+        }
+
+    }
+
+    @Override
+    public int getID() {
+        // TODO Auto-generated method stub
```
Comment (Pathemeous) [inline src/main/java/nl/tudelft/semgroup4/GameEndedState.java]: Might want to delete this TODO.

< layout

File: src/main/java/nl/tudelft/semgroup4/util/HighscoresHelper.java [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+import java.nio.file.OpenOption;
+import java.nio.file.Paths;
+import java.nio.file.StandardOpenOption;
+import java.util.ArrayList;
+import java.util.List;
+
+import org.json.JSONArray;
+import org.json.JSONObject;
+
+/**
```
Comment (Pathemeous) [inline src/main/java/nl/tudelft/semgroup4/util/HighscoresHelper.java]: Could you add the reason for this class to the javaDoc?

< documentation

**Comment by Pathemeous** [comment]
Could you maybe add JavaDoc to the Classes that you created and all methods? Class JavaDoc is useful for knowing whether the class conforms to the single-responsibility principle.

< documentation

File: src/main/java/nl/tudelft/semgroup4/GameEndedState.java (hunk 1 above)
Comment (justinvdk) [inline src/main/java/nl/tudelft/semgroup4/GameEndedState.java]: This was there already, but i will remove it < layout

**Comment by Pathemeous** [comment]
Thanks, that makes me content

**Pull request merged**